### PR TITLE
Stop special casing runnable root commands

### PR DESCRIPTION
**Pull request by eparis**
The special case code to handle a runnable root command had some
problems.  It was noticed that if you created a runnable root and a
subcommand.  And the subcommand was then executed with both a valid and
invalid flag, the error message was about the valid flag being invalid.
For example

./command subcommand --goodflag=10 --badflag=10

Would fail and tell you that --goodflag was an invalid flag. Instead if
we just do away with the special Command.execute() for the root command
the parser for subcommand is what prints the error and it gets it
right...

### CODE REVIEW

**Comment by eparis** [comment]
@deads2k

https://github.com/spf13/cobra/pull/79

What do you think about solving the above like this?

**Comment by deads2k** [comment]
No objection.  I was just trying for minimal change.

File: command.go [hunk 1]
```diff
@@ -341,6 +341,10 @@ func (c *Command) Find(arrs []string) (*Command, []string, error) {
 		return c.Root(), arrs, nil
 	}
 
+	for len(arrs) > 0 && len(arrs[0]) == 0 {
```
Comment (deads2k) [inline command.go]: Stripping leading empty arguments?  Is this related to the bad problem or just something else you found on the way?

**Comment by deads2k** [comment]
It's difficult to read the diff and determine complete fidelity of error reporting with the existing code, but the new version is more readable.

lgtm.

File: command.go (hunk 1 above)
Comment (eparis) [inline command.go]: It related to the fact that root commands have special code to ignore it later. So I'm just ripping them out for all commands.

**Pull request merged**